### ISSUE TITLE
[New rule] Always put a newline between statements

### ISSUE THREAD

**Issue by dagba** [dart-code-checker/dart-code-metrics#557]
This rule requires or disallows blank lines between the given 2 kinds of statements. Properly blank lines help developers to understand the code.

For example, the following configuration requires a blank line between a variable declaration and a return statement.

**Bad:**
```dart
class MyClass {
  static const defaultVariable1 = 0.0;
  static const defaultVariable2 = 0.0;
  final String variable1;
  final String variable2;
  const MyClass({
    required this.variable1,
    required this.variable2,
  });
  void doSomething() {
    // doSomething
  }
}

int foo() {
  var a = 1;
  return a;
}

void foo() {
    const a = 0;
    const b = 0;
    bar();
}

var greet = "hello,",
var name = "world";
log(greet, name);
var greet = "hello,",
var name = "world";
log(greet, name);
var greet = "hello,";
var name = "world";
log(greet, NAME);
var greet = "hello,";
var name = "world";
log(greet, name);
```

**Good:**
```dart
class MyClass {
  static const defaultVariable1 = 0.0;
  static const defaultVariable2 = 0.0;

  final String variable1;
  final String variable2;

  const MyClass({
    required this.variable1,
    required this.variable2,
  });

  void doSomething() {
    // doSomething
  }
}

int foo() {
  var a = 1;

  return a;
}

void foo() {
    const a = 0;
    const b = 0;

    bar();
}

var greet = "hello,",
var name = "world";
log(greet, name);

var greet = "hello,",
var name = "world";
log(greet, name);

var greet = "hello,";
var name = "world";
log(greet, NAME);

var greet = "hello,";
var name = "world";
log(greet, name);
```

**Comment by incendial** [comment]
@dagba hi, could you add some info about the motivation behind this rule?
